src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1232,6 +1239,37 @@ def _get_log(url, user_name, password, log_file=None):
                 if not data:
                     break
                 f.write(data)
+    elif number_of_lines:  # print last 'n' number of lines
+        api_str = r.read().decode(encoding='utf-8')
+
+        href_count = api_str.count('https')
+
+        if href_count > 1:
+            logger.info('Will show last %s lines of logs per instance', number_of_lines)
+
+        for x in range(0, href_count):  # pylint: disable=unused-variable
+            index_of_path = api_str.index('\",\"path')
+            href_str = api_str[api_str.index("https"):index_of_path]
+            size = api_str[api_str.index('\"size\":') + len('\"size\":'):api_str.index(',\"href\"')]
+            logger.info('\nUsing endpoint: %s', href_str)
+            url = href_str
+            if int(size) > 512000:
+                logger.warning('Size of logs:%s bytes, will fetch only last 512000 bytes', size)
+                headers.update({'Range': 'bytes=-512000'})  # only get the last 512k
+
+            r = http.request(
+                'GET',
+                url,
+                headers=headers,
+                preload_content=False
+            )
+            lines = (r.data.decode(encoding='utf-8', errors='replace')
+                     .encode(std_encoding, errors='replace')
+                     .decode(std_encoding, errors='replace')
+                     .replace('\\n', '\r\n'))
```

Comment:
I don't think this is correct, as `\r\n` is windows only

Reply:
Again, can you share what the response payload look like? Also the client side parsing logic looks a bit excessive, so I would suggest you should write some unit tests

Reply:
Will change to `\n`

r.data may look something like

b'\xef\xbb\xbf2017-11-07 19:26:06.974 INFO  - Issuing docker pull: imagename =myrepo/myimage:v-193806a\n2017-11-07 19:26:07.264 INFO  - Issuing docker pull: imagename =myrepo/myimage:v-193806a\n2017-11-07 19:26:07.686 INFO  - Issuing docker pull myrepo/myimage:v-193806a\n2017-11-07 19:26:08.493 INFO  - docker pull returned STDOUT>> Pulling repository myrepo/myimage\n\n2017-11-07 19:26:08.494 ERROR - docker pull returned STDERR>> Error: image myimage:v-193806a not found\n\n'

Sometimes I see `\\n`